metronome/config/specs/src/io/iohk/metronome/config/ConfigParserSpec.scala

```diff
@@ -175,6 +175,41 @@ class ConfigParserSpec
     }
   }
 
+  it should "deal with escape characters according to rfc7159 (section-7)" in {
+    val conf = ConfigFactory.parseString(
+      """{"quotation_mark":"",
+        |"reverse_solidus":"",
+        |"solidus":"",
+        |"backspace":"",
+        |"form_feed":"",
+        |"line_feed":"",
+        |"carriage_return":"",
+        |"tab":"",
+        |"java_home":"C:\\Program Files\\java\\jdk-8\\bin"}""".stripMargin
+    )
+    val orig = ConfigParser.toJson(conf.root())
+    val json = ConfigParser.withCamelCase(orig)
+
+    val env = Map(
+      "QUOTATION_MARK"  -> """"""",
+      "REVERSE_SOLIDUS" -> """\""",
+      "SOLIDUS"         -> """/""",
+      "BACKSPACE"       -> """\b""",
+      "FORM_FEED"       -> """\f""",
+      "LINE_FEED"       -> """\n""",
+      "CARRIAGE_RETURN" -> """\r""",
+      "TAB"             -> """\t""",
```

Comment:
These examples aren't quite right. You see if you want to actually place a newline in a string, you should not use triple quotes.

```shell
scala> println(""" foo \n bar """)
 foo \n bar 

scala> println(" foo \n bar ")
 foo 
 bar 
```

The point of triple quotes is that we don't have to escape special characters inside them, they are as-is. 

When you see this JSON: `"lineFeed":"\\n"` that's not a line feed but the characters `\` and `n` in a string. A JSON string with a line feed would look like `"lineFeed":"\n"`. 

That was the error we got: `"javaHome": "C:\Program Files\Java"` had `\P` which looked like an escape of a special character, but `\P` doesn't exist. So to have special chars, they should not be escaped, hence no need to triple quote them, and no need to escape in the assertion either.

Reply:
That's true, I overlooked it.
Regarding the assertion, I keep the message such string with a triple quote. Because the idea is to scape the special characters, and that is working and testing with this use case.
Thanks for note this important point